unintended_ml_bias/new_madlibber/madlibber.py

```diff
@@ -0,0 +1,98 @@
+import re
+import csv
+
+from enums import Toxicity, WordType, WordConnotation, WordGender
+
+class Madlibber(object):
+  def __init__(self, path_helper):
+    self.path_helper = path_helper
+    self.__words = {}
+
+  def load_and_sanity_check_words(self):
+    f = open(self.path_helper.word_file, 'r')
+    csv_f = csv.reader(f)
+    next(csv_f)  # skip header
+    count = 0
+    for line in csv_f:
+      word_type, word_conn, word_gender, word = line
+
+      if word == '':
+        continue
+
+      if not self.__sanity_check(word_type, WordType):
+        raise ValueError("'{}' is not a valid word type".format(word_type))
+
+      if not self.__sanity_check(word_conn, WordConnotation):
+        raise ValueError("'{}' is not a valid word connotation".format(word_connotation))
+
+      if not self.__sanity_check(word_gender, WordGender):
+        raise ValueError("'{}' is not a valid word gender".format(word_gender))
+
+      unicode_word = word.decode('utf-8')
+      word_key = self.get_word_key(word_type, word_conn, word_gender)
+      self.__words.setdefault(word_key, [])
+      self.__words[word_key].append(unicode_word)
+      count += 1
+    f.close()
+
+    print("Loaded {} words".format(count))
+    print("Loaded {} word keys".format(len(self.__words.keys())))
+    print("Loaded word keys: {}".format(", ".join(["'{}'".format(k) for k in self.__words.keys()])))
+
+  def fill_templates(self):
+    f = open(self.path_helper.sentence_template_file, 'r')
+    csv_f = csv.reader(f)
+    next(csv_f)  # skip header
+
+    fout = open(self.path_helper.output_file, 'w')
+    csv_fout = csv.writer(fout)
+    csv_fout.writerow(['template','toxicity','phrase'])
+
+    count = 0
+    for line in csv_f:
```

Comment:
Is it possible to add a feature to calculate and print the expected number of output sentences before diving in to create them all? It's super easy to create templates and terms that generate extremely large sets that aren't actually useful and if the set is large this code can take some time to run. Printing the expected number of sentences would give the user a better expectation of how long this will take and let them know if they've created an unreasonable set.

Reply:
Sure, I will work on adding this.